packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -89,60 +111,154 @@ export const prepareLiquidityPoolKit = (zone, zcf, USDC, tools) => {
+            outstandingLends,
+            poolStats,
           );
-          this.state.shareWorth = withFees(shareWorth, amount);
-          external.publishShareWorth();
+
+          // COMMIT POINT
+          try {
+            zcf.atomicRearrange(
+              harden([[poolSeat, assetManagerSeat, amounts]]),
+            );
+          } catch (cause) {
+            const reason = Error('🚨 cannot commit borrow', { cause });
+            console.error(reason.message, cause);
+            zcf.shutdownWithFailure(reason);
+          }
+
+          Object.assign(this.state, post);
+          this.facets.external.publishPoolMetrics();
         },
+        // TODO repay failed `LOA.deposit()`
       },
+      repayer: {
+        /**
+         * @param {ZCFSeat} assetManagerSeat
+         * @param {RepayAmountKWR} amounts
+         */
+        repay(assetManagerSeat, amounts) {
+          const {
+            contractSeat,
+            outstandingLends,
+            poolSeat,
+            poolStats,
+            shareWorth,
+          } = this.state;
+          checkPoolBalance(poolSeat, shareWorth, usdc.brand, outstandingLends);
+          // determine new shareWorth, outstandingLends, and poolMetrics
+          const post = repayCalc(
+            shareWorth,
+            amounts,
+            outstandingLends,
+            poolStats,
+          );
+
+          const { ContractFee, ...rest } = amounts;
+
+          // COMMIT POINT
+          try {
+            zcf.atomicRearrange(
+              harden([
+                [
+                  assetManagerSeat,
+                  poolSeat,
+                  rest,
+                  { USDC: add(amounts.PoolFee, amounts.Principal) },
+                ],
+                [
+                  assetManagerSeat,
+                  contractSeat,
+                  { ContractFee },
+                  { USDC: ContractFee },
+                ],
+              ]),
+            );
+          } catch (cause) {
+            const reason = Error('🚨 cannot commit repay', { cause });
+            console.error(reason.message, cause);
+            zcf.shutdownWithFailure(reason);
+          }
 
+          Object.assign(this.state, post);
+          this.facets.external.publishPoolMetrics();
+        },
+      },
       external: {
-        publishShareWorth() {
-          const { shareWorth } = this.state;
-          const { recorder } = this.state.shareWorthRecorderKit;
+        publishPoolMetrics() {
+          const { poolStats, shareWorth, poolSeat } = this.state;
+          const { recorder } = this.state.poolMetricsRecorderKit;
           // Consumers of this .write() are off-chain / outside the VM.
           // And there's no way to recover from a failed write.
           // So don't await.
-          void recorder.write(shareWorth);
+          void recorder.write(
+            /** @type {PoolMetrics} */ ({
+              availableBalance: poolSeat.getAmountAllocated('USDC', usdc.brand),
```

Comment:
I wonder if that's a slight misnomer. If the poolSeat allocation is 1 USDC, you can't borrow all of it. The poolSeat can never go empty.

Reply:
Per our convo, we'll publish `encumberedBalance` in `PoolMetrics` instead of `availableBalance`